lib/util.js

```diff
@@ -324,8 +324,20 @@ module.exports = {
             currentNode.addChildren(currentPath[i], new Node({
               name: currentPath[i],
               requestCount: 0,
-              type: 'item-group'
+              requests: [],
+              children: {},
+              type: 'item-group',
+              childCount: 0
             }));
+
+            /* eslint-disable max-depth */
+            if (!currentNode.childCount) {
```

Comment:
@Dhroov7 need to explain this. Why are we adding to `childCount` here? Also, can we make sure that childCount is always initialized to 0? That'll prevent the need for the `if` check.

Reply:
We cannot remove the check because it will break in the first iteration.

Reply:
childCount should be added as a new property in the Node object, and initialized to 0. Check `trie.js`

Reply:
@abhijitkane Oh yeah..added childCount property in Node object.